source/code/plugins/in_npmd_server.rb

```diff
@@ -0,0 +1,428 @@
+                            Logger::logWarn "No valid data items found in sent json #{_json}", Logger::loop
+                        else
+                            _uploadData = _json["DataItems"].reject {|x| x["SubType"] == "ErrorLog"}
+                            _errorLogs  = _json["DataItems"].select {|x| x["SubType"] == "ErrorLog"}
+                            _uploadData.each do |item|
+                                if item.key?("SubType")
+                                    # Append FQDN to path data
+                                    if !@fqdn.nil? and item["SubType"] == "NetworkPath"
+                                        @num_path_data += 1 unless @num_path_data.nil?
+                                        item["Computer"] = @fqdn
+                                        # Append agent Guid to agent data
+                                    elsif !@agentId.nil? and item["SubType"] == "NetworkAgent"
+                                        @num_agent_data += 1 unless @num_agent_data.nil?
+                                        item["AgentId"] = @agentId
+                                    end
+                                end
+                            end
+                            emit_upload_data_dataitems(_uploadData) if !_uploadData.nil? and !_uploadData.empty?
+                            emit_error_log_dataitems(_errorLogs) if !_errorLogs.nil? and !_errorLogs.empty?
+                        end
+                    end
+                rescue StandardError => e
+                    log_error "Got error while reading data from NPMD: #{e}", Logger::loop + Logger::resc
+                end
+            end while !@npmdClientSock.nil?
+        end
+
+        def package_and_send_error_log(msg)
+            _dataItems = Array.new
+            _h = Hash.new
+            _h["Message"] = msg
+            _dataItems << _h
+            emit_error_log_dataitems(_dataItems)
+        end
+
+        def emit_error_log_dataitems(dataitems)
+            _record = Hash.new
+            _record["DataType"] = "HEALTH_ASSESSMENT_BLOB"
+            _record["IPName"]   = "LogManagement"
+            _record["DataItems"] = dataitems
+            router.emit(@tag, Engine.now, _record)
+        end
+
+        def emit_upload_data_dataitems(dataitems)
+            _record = Hash.new
+            _record["DataType"] = "NETWORK_MONITORING_BLOB"
+            _record["IPName"]   = "NetworkMonitoring"
+            _record["DataItems"] = dataitems
+            router.emit(@tag, Engine.now, _record)
+        end
+
+        def check_and_get_guid
+            create_new_guid = true
+            if File.exist?@location_uuid_file
+                f = File.new(@location_uuid_file, "r")
+                @agentId = f.read
+                f.close
+
+                create_new_guid = false if (!@agentId.nil? and !@agentId.empty?)
+            end
+
+            if create_new_guid
+                @agentId = SecureRandom.uuid
```

Comment:
We are introducing a new UUID rather than the omsagent id. We plan to have one UUID right now for the NPMD solution over multiple restarts unless there is a purge.